src/main/java/frc/robot/Robot.java

```diff
@@ -4,68 +4,258 @@
+    private Gripper gripper;
+    private OI oi;
+    private ArmGravityCompensation compensation;
+    private VisionService vision;
+    private LedsService leds;
+    private AutoChooser autoChooser;
+
+    /**
+     * A command that sets all the motor controllers to coast.
+     */
+    private WrapperCommand userCommand;
+
+    @Override
+    public void robotInit() {
+        getInstances();
+        setCompressor();
+        setDefaultJointsCommands();
+        setNamespaceTestingCommands();
+        autoChooser = new AutoChooser(
+                new RootNamespace("auto chooser"),
+                new PlanBWindow(drivetrain).getCommand(), "plan b window",
+                new PlanBEdge(drivetrain).getCommand(), "plan b edge",
+                new SplooshAndVamooseWindow(drivetrain).getCommand(), "sploosh and vamoose",
+                new SmashAndDash(drivetrain).getCommand(), "smash and dash"
+        );
+        firstJoint.configureEncoders();
+        secondJoint.configureEncoders();
+        vision.setBackLimelightPipeline(VisionService.LimelightPipeline.HIGH_RRT);
+        vision.setFrontLimelightPipeline(VisionService.LimelightPipeline.HIGH_RRT);
+        userCommand = new InstantCommand(() -> {
+            drivetrain.setMode(CANSparkMax.IdleMode.kCoast);
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kCoast);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kCoast);
+        }).ignoringDisable(true);
+    }
+
+    @Override
+    public void robotPeriodic() {
+        CommandScheduler.getInstance().run();
+        namespace.update();
+        drivetrain.periodic();
+        firstJoint.periodic();
+        secondJoint.periodic();
+        gripper.periodic();
+        vision.periodic();
+        leds.periodic();
+        SmashAndDash.update();
+        if (RobotController.getUserButton()) {
+            userCommand.schedule();
+        }
+    }
 
-  }
+    @Override
+    public void disabledInit() {
+        CommandScheduler.getInstance().cancelAll();
+        firstJoint.finish();
+        secondJoint.finish();
+        drivetrain.finish();
+        new InstantCommand(() -> {
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        }).ignoringDisable(true).schedule();
+        new OpenGripper(gripper).ignoringDisable(true).schedule();
+    }
 
-  @Override
-  public void robotPeriodic() {
-    CommandScheduler.getInstance().run();
-  }
+    @Override
+    public void disabledPeriodic() {
 
-  @Override
-  public void disabledInit() {
+    }
 
-  }
+    @Override
+    public void autonomousInit() {
+        new InstantCommand(() -> {
+            firstJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+            secondJoint.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        });
+        new CloseGripper(gripper).schedule();
+        CommandBase auto = null;
+        auto = new SmashAndDash(drivetrain).getCommand();
```

Comment:
You might want to leave it as a comment (don't want anybody to kill themselves when accidentally running autonomous, right?)

Reply:
that would be funny though